### PR TITLE
Clarification for Shell and Distribution Tutorials 

### LINKED ISSUE #44: Notes from @vassiki
# Shell
- [ ] `pager` cmdline is debian-specific.  In general it is `PAGER` env variable which instructs which pager to use.  Also connection to the `man` command was not clear
- [ ] in shortcuts describe `n` for search
- [ ] folks who run `man -k containers` on their systems might find none.  Change to `shell`?
- [ ] show the value of the `$PATH` when talking about it; describe the role of `:` in there (may be note that on Windows it is not `:` but `;`)
- [ ] `-a` for which is kinda buried, may be also ask to try it also on `python` on their laptops
- [ ] note about needing two `TABS` when completion is ambiguous (multiple choices)
- [ ] note for poor Mac users that Alt doesn't translate to Meta and they need to use Esc for e.g. Alt-.
- [ ] might be worth noting `Ctrl-C` as the way to cancel and get out of incremental search or editing current line, while still leaving it on the screen
- [ ] Fail early:  not yet clear what to do with it
      - may be make analogy to Python, where script would interrupt as soon as error or undefined variable is used
      - the whole section is a bit "over the head" because, probably, we do not do much of scripting there
      - not clear where to put `set -e`
- [ ] may be consider "pre feeding bash history" with some real data analysis example (e.g. bet) which ppl need to place into a script.

### PR DESCRIPTION
**Pull request by vassiki**
All points addressed except for the last two. 

Closes #44

### CODE REVIEW

**Comment by vassiki** [comment]
Added fixes addressing #45 

But no exercises added for git-annex

**Comment by yarikoptic** [comment]
awesome! thank you @vassiki !  I will give it a closer look tomorrow before merging.  If you would like to keep working on it for a bit more, just change title to start with "WIP: " so I know not to merge just yet

File: _episodes/02-01-shell.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ questions:
 objectives:
 - "Provide hints on efficient use of the collected shell history of commands"
 - "Explain how to make shell scripts more robust and less dangerous"
-keypoints:
+keypoints: 
 - "A command line shell is a powerful tool and learning additional
```
Comment (yarikoptic) [inline _episodes/02-01-shell.md]: Typically before doing actual commit I run `git diff` to inspect my changes, and then possibly adjust so I do not introduce unrelated changes, e.g. such a trailing space.

File: _episodes/02-01-shell.md [hunk 2]
```diff
@@ -72,15 +72,16 @@ The `man` command provides access to manpages available for many (if
 not the majority) of the available commands.  Manpages often provide a
 very detailed description and consist of many pages of textual
 documentation.  It gets presented to you in a `pager`---a basic command
-for viewing and navigation of the text file.  The most common pages are
+for viewing and navigation of the text file. The `pager` to be used is specified
+as an environment variable, two common examples of which are
```
Comment (yarikoptic) [inline _episodes/02-01-shell.md]: please specify which environment variable (`$PAGER`)

File: _episodes/02-01-shell.md [hunk 3]
```diff
@@ -143,6 +161,15 @@ impact what external commands and libraries you are using.
 >
 > What about the `python` command?   Try `which -a` as well.
 >
+> ~~~
+> % which -a python
+> ~~~
+> {: .bash}
+>
+> `which -a` is a neat way to see all the versions of a 
+> package you across paths in the environment, listed
```
Comment (yarikoptic) [inline _episodes/02-01-shell.md]: smth like "have available" is missing after "you"?

File: _episodes/02-01-shell.md [hunk 4]
```diff
@@ -143,6 +161,15 @@ impact what external commands and libraries you are using.
...
+> in the order of path specified.
```
Comment (yarikoptic) [inline _episodes/02-01-shell.md]: "in the order of paths specified in `$PATH` environment variable."

File: _episodes/02-02-distros.md [hunk 5]
```diff
@@ -104,6 +104,8 @@ the instructions on http://neuro.debian.net .
 > > ~~~
 > > {: .bash}
 > {: .solution}
+For those using older VM images for Neurodebian, you might have to use `apt-cache policy`
```
Comment (yarikoptic) [inline _episodes/02-02-distros.md]: We proudly use two capitals in NeuroDebian ;)

File: _episodes/02-02-distros.md [hunk 6]
```diff
@@ -301,8 +303,9 @@ works with a complete and annotated history of content changes.
 > > {: .bash}
 > > But SHA1 is not "ordered", i.e. from observing one SHA1 you cannot tell
 > > if it comes later or earlier in development of the content.
-> > [git tag](https://git-scm.com/docs/git-tag) allow to "tag" specific content
-> > versions with meaningful and/or comparable version strings.  Run `git tag` to
+> > [git tag](https://git-scm.com/docs/git-tag) allows one to "tag" specific content
+> > versions with meaningful and/or comparable version strings, enabling the process of
+> > benchmarking progress in a project. Run `git tag` to
```
Comment (yarikoptic) [inline _episodes/02-02-distros.md]: hm, "benchmarking progress"... not sure

File: _episodes/02-02-distros.md [hunk 7]
```diff
@@ -331,7 +334,9 @@ works with a complete and annotated history of content changes.
     of your current "version" of the repository
   - commands such as [git push], [git fetch], [git pull], etc exchange
     references (tags, branches, etc) and the content they point to
-    between clones of the repository
+    between clones of the repository. Useful rule of thumb: [git pull] = [git fetch] + [git merge]
+    Before [GitHub], [git fetch] was the only way to check differences
+    between local and remote version of a repository.
```
Comment (yarikoptic) [inline _episodes/02-02-distros.md]: correct, but `git fetch` itself doesn't check difference, it just fetches changes. so may be expand wording a bit to reflect that?

**Comment by vassiki** [comment]
Hi @yarikoptic , please review the fixes to the comments you made here!
